src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -306,15 +324,13 @@ void Terminal::_ExpandDoubleClickSelectionRight(const COORD position)
         cellChar = _buffer->GetCellDataAt(positionWithOffsets)->Chars();
     }
 
-    if (positionWithOffsets.X != bufferViewport.RightInclusive() || _isWordDelimiter(cellChar))
+    if (positionWithOffsets.X != bufferViewport.RightInclusive() && _isWordDelimiter(cellChar))
     {
         // move off of delimiter to highlight properly
         bufferViewport.DecrementInBounds(positionWithOffsets);
     }
 
-    THROW_IF_FAILED(ShortSub(positionWithOffsets.Y, gsl::narrow<SHORT>(_ViewStartIndex()), &positionWithOffsets.Y));
-    _endSelectionPosition = positionWithOffsets;
-    _endSelectionPosition_YOffset = gsl::narrow<SHORT>(_ViewStartIndex());

```

Comment:
i think this is no longer being set in the general case? does it matter? wait, the _position offset_ should be the same for both the start AND end of the selection.. because they're in the same viewport

Reply:
Moved this logic to `DoubleClickSelection()` so that we can reuse this function at render time. We don't want to update the offsets at render time, only when the double click was first created.